torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -472,14 +472,18 @@ def return_future():
 
 
 class RpcTest(RpcAgentTestFixture):
-    def _skip_if_tensorpipe_agent(old_func):  # noqa: B902
-        def decorator(self):
-            return unittest.skipIf(
-                self.rpc_backend == rpc.backend_registry.BackendType.TENSORPIPE,
-                "This test is not yet supported in the Tensorpipe Agent"
-            )(old_func)
+    def _skip_if_tensorpipe_agent(test_func):  # noqa: B902
+        @wraps(test_func)
+        def wrapped_test_func(self):
+            if self.rpc_backend == rpc.backend_registry.BackendType.TENSORPIPE:
+                # In theory we should call self.skipTest(reason), but this does
+                # not play well with the test runner, which sees an exception
+                # being raised and marks the test as failed. Thus instead we
+                # make the test a successful no-op.
+                return
```

Comment:
Do we need to call `sys.exit()` and provide an skip code here?

Reply:
Do you mean these?
https://github.com/pytorch/pytorch/blob/1a74bb84f2b0fd0e1a532941645e24e2cfea1e6a/torch/testing/_internal/common_distributed.py#L26-L33

I wasn't aware of these. We could in fact add a new code just for those. Note however that this decorator will go away later in the stack (in #40819). So I don't know if it makes sense to invest that much into it now just to then undo it all.

Reply:
Oh I see. Haven't reached #40819 yet. In that case, please ignore this comment.